xls/dslx/tests/std_find_index_on_enum_array.x

```diff
@@ -0,0 +1,49 @@
+// Copyright 2025 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Validates that we can find the index of a value in an enum array.
```

Comment:
Is `//!` for some reason I'm unaware of?

Reply:
For rustdoc this means "it's a comment block associated with the module it's enclosed in" -- we don't have a dslxdoc yet but I hope to fix that one day so just trying to stick to what we'd tend to do in Rust where possible.